File: solutions/203.remove-linked-list-elements/iteration.ts
/*
 * @lc app=leetcode.cn id=203 lang=typescript
 *
 * https://leetcode-cn.com/problems/remove-linked-list-elements
 */

/**
 * Definition for singly-linked list.
 */
class ListNode {
    val: number;
    next: ListNode | null;
    constructor(val?: number, next?: ListNode | null) {
        this.val = val === undefined ? 0 : val;
        this.next = next === undefined ? null : next;
    }
}

// @lc code=start
function removeElements(head: ListNode | null, val: number): ListNode | null {
    const dummy = new ListNode(0, head);
    let current = dummy;
    while (current.next) {
        if (current.next.val === val) {
            current.next = current.next.next;
        } else {
            current = current.next;
        }
    }
    return dummy.next;
}
// @lc code=end

export {};

File: solution/27.remove-element/iteration.rs
/*
 * @lc app=leetcode.cn id=27 lang=rust
 *
 * https://leetcode-cn.com/problems/remove-element
 */

// @lc code=start
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut i = 0;
        for j in 0..nums.len() {
            if nums[j] != val {
                nums[i] = nums[j];
                i += 1;
            }
        }
        i as i32
    }
}
// @lc code=end
